spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/dataAdapter/GeoPackageAdapter.java

```diff
@@ -47,114 +81,552 @@
+                // if there are no layer names supplied in the query filter, then search on all tables
+                List<String> layerNames = queryFilter.getLayerIds();
+                if (layerNames.size() == 0) {
+                    layerNames = getFeatureSourceNames();
+                }
+                Observable.from(layerNames)
+                        .filter(new Func1<String, Boolean>() {
+                            @Override
+                            public Boolean call(String layer) {
+                                return sources.contains(getFeatureSourceByName(layer));
+                            }
+                        })
+                        .flatMap(new Func1<String, Observable<SCSpatialFeature>>() {
+                            @Override
+                            public Observable<SCSpatialFeature> call(final String layerName) {
+                                final SCGpkgFeatureSource source = getFeatureSourceByName(layerName);
+                                return gpkg.db.createQuery(
+                                        layerName,
+                                        String.format(
+                                                "SELECT %s FROM %s WHERE %s IN (%s) LIMIT %d",
+                                                getSelectColumnsString(source),
+                                                layerName,
+                                                source.getPrimaryKeyName(),
+                                                createRtreeSubQuery(source, queryFilter.getPredicate().getBoundingBox()),
+                                                100
+                                        )
+                                ).flatMap(getFeatureMapper(source));
+                            }
+                        })
+                        .subscribe(new Action1<SCSpatialFeature>() {
+                            @Override
+                            public void call(SCSpatialFeature feature) {
+                                subscriber.onNext(feature);
+                            }
+                        });
             }
         });
     }
 
+    /**
+     * Query for a feature by its {@link SCKeyTuple}.
+     *
+     * @param keyTuple
+     * @return
+     */
+    public Observable queryById(final SCKeyTuple keyTuple) {
+        final String tableName = keyTuple.getLayerId();
+        final SCGpkgFeatureSource featureSource = getFeatureSourceByName(tableName);
+        if (featureSource == null) {
+            return Observable.error(
+                    new SCDataStoreException(
+                            SCDataStoreException.ExceptionType.LAYER_NOT_FOUND,
+                            String.format("%s was not a valid feature table name.", tableName)
+                    )
+            );
+        }
+        else {
+            return Observable.create(new Observable.OnSubscribe<SCSpatialFeature>() {
```

Comment:
Don't need to create a new observable if you aren't doing any transformation in the subscribe